src/test/java/io/thinkit/edc/client/connector/ContractDefinitionsTest.java

```diff
@@ -39,6 +41,27 @@ void should_get_a_contract_definition() {
         assertThat(contractDefinition.getContent().createdAt()).isGreaterThan(0);
     }
 
+    @Test
+    void should_get_a_contract_definition_async() {
+        try {
+            Result<ContractDefinition> contractDefinition =
+                    contractDefinitions.getAsync("definition-id").get();
```

Comment:
instead of using `.get()`, assertj offers a special assertion for futures, and this test can be represented as:
```
assertThat(contractDefinitions.getAsync("123")).succeedsWithin(5, TimeUnit.SECONDS)
                    .satisfies(contractDefinition -> {
        
                        assertThat(contractDefinition.isSucceeded()).isTrue();
                        assertThat(contractDefinition.getContent().id()).isNotBlank();
                        assertThat(contractDefinition.getContent().accessPolicyId())
                                .isNotNull()
                                .satisfies(accessPolicyId -> assertThat(accessPolicyId).isEqualTo("asset-policy-id"));
                        assertThat(contractDefinition.getContent().contractPolicyId())
                                .isNotNull()
                                .satisfies(contractPolicyId -> assertThat(contractPolicyId).isEqualTo("contract-policy-id"));
                        assertThat(contractDefinition.getContent().createdAt()).isGreaterThan(0);
                    });
```